### PR TITLE
Carson Jardine IC 2

### PR DESCRIPTION
**Pull request by carson-jardine**

### CODE REVIEW

File: lib/house.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  attr_reader :address, :rooms
+
+  def initialize(price, address)
+    @price = price
+    @address = address
+    @rooms = []
+  end
+
+  def price
+    @price.delete("$").to_i
```
Comment (mikedao) [inline lib/house.rb:11]: Great

File: lib/house.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    rooms.sum do |room|
+      room.area
+    end
+  end
+
+  def details
+    details = {
+      "price" => price,
+      "address" => address
+    }
```
Comment (mikedao) [inline lib/house.rb:38]: Very clean.

File: lib/house.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  end
+
+  def price_per_square_foot
+    (price.to_f / area).round(2)
+  end
+
+  def rooms_sorted_by_area
+    rooms.sort_by do |room|
+      room.area
```
Comment (mikedao) [inline lib/house.rb:47]: I think we wanted this sorted in the other direction.

File: lib/house.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  def rooms_by_category
+    rooms_by_category = {}
+
+    categories = rooms.map do |room|
+      room.category
+    end
+    categories.uniq.each do |category|
+      rooms_by_category[category] = rooms_from_category(category)
+    end
+    rooms_by_category
```
Comment (mikedao) [inline lib/house.rb:60]: This is doing a bit much how can we break this down more?

File: lib/house.rb (hunk 3 above)
Comment (carson-jardine) [reply lib/house.rb:47]: @mikedao shoot, I need to pay better attention to the iteration pattern. I would throw a reverse on there.

File: lib/house.rb (hunk 4 above)
Comment (carson-jardine) [reply lib/house.rb:60]: @mikedao I realized I could use group_by about 20 minutes after we had to turn these in lol, I would do the following: 
```
def rooms_by_category
rooms.group_by do |room|
room.category
end 
end 
```

**Pull request closed without merging**